### PR TITLE
Use feature flag for Intercom rollout

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

Rely on the new feature flag system over `isDevelopmentOrDustWorkspace` on the Intercom connection.

## Risk

Low, rollback is easy.

## Deploy Plan

Nothing special.

### CODE REVIEW

File: front/lib/api/assistant/global_agents.ts [hunk 1]
```diff
@@ -825,8 +825,11 @@ export async function getGlobalAgents(
     )
   );
 
-  // ROLLOUT INTERCOM
-  if (!isDevelopmentOrDustWorkspace(owner)) {
+  const isIntercomEnabled = await isFeatureEnabled(
```
Comment (fontanierh) [inline front/lib/api/assistant/global_agents.ts:828]: any chance we can Promise.all this guy with something above ?

File: front/pages/w/[wId]/builder/data-sources/managed.tsx [hunk 2]
```diff
@@ -518,8 +521,7 @@ export default function DataSourcesView({
             .map((ds) => {
               const isBuilt =
                 ds.status === "built" ||
-                (ds.status === "built-dust-only" &&
-                  isDevelopmentOrDustWorkspace(owner));
+                (ds.connectorProvider === "intercom" && isIntercomEnabled);
```
Comment (fontanierh) [inline front/pages/w/[wId]/builder/data-sources/managed.tsx:524]: Could potentially have this be slightly more generic by having the status be "built" but have an optional "feature_flag" field on the CONNECTOR_CONFIGURATIONS.

If the FF field is non-null, then we check if the FF is enabled for the workspace. WDYT ?

**Review by fontanierh — APPROVED** [review]
nothing blocking just a small suggestion

File: front/lib/api/assistant/global_agents.ts (hunk 1 above)
Comment (PopDaph) [reply front/lib/api/assistant/global_agents.ts:820]: I added a commit to filter before calling `getGlobalAgent` for intercom, but didn't promise all with something above. Seems weird to promise the intercom feature flag with the fetching of prod credentials or all datasources, no? Happy to do it anyway if you think it's better

File: front/pages/w/[wId]/builder/data-sources/managed.tsx (hunk 2 above)
Comment (PopDaph) [reply front/pages/w/[wId]/builder/data-sources/managed.tsx]: I added a commit to be more generic, but with default to preview to make sure we never roll out by mistake. WDYT?

Comment (fontanierh) [reply front/pages/w/[wId]/builder/data-sources/managed.tsx]: Sounds good, there's a tiny amount of type-safety we could add to make sure that if `preview`, `rollingOutFlag` has to be non-null (and the other way around), but this is very nit :)

File: front/lib/api/assistant/global_agents.ts (hunk 1 above)
Comment (fontanierh) [reply front/lib/api/assistant/global_agents.ts:820]: Not a strong opinion, but IMO if we can fetch stuff concurrently then it tends to be better to reduce latency. really nit and totally fine with the code as-is 👍

Comment (PopDaph) [reply front/lib/api/assistant/global_agents.ts]: Refactored to use owner.flags!

**Pull request merged**